### PR TITLE
KM544 - add terraform for setting up OIDC access to ECS

### PR DESCRIPTION
**Pull request by haavardeide**
This adds support for OIDC authentication between AWS and github in the workflow action. Removing the usage of long-lived access keys for pushing image to AWS ECR 

The `tf/` code have been applied to `dev`, see `okctl-reference-app-github-ecr-push` under IAM.

The workflow have been executed on this branch - and resulted in
* a new image pushed to AWS ECS, see `okctl-reference-app` repo
* IAC repo updated with the new image, see https://github.com/oslokommune/okctl-reference-iac/commit/8ee94ea70ae7237619eb1ca2c89bd7b84de14692

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,3 +21,38 @@ Run tests
...
+* `repository`: The {organization}/{okct-config} repo containing your okctl IAC code
+* `ref`: branch name you are using (main/master) for your IAC repo
+* `DEPLOYMENT_YAML_FILE`: path to your application deployment patch
+* The git commit message at the end of `jobs.update-tag`
+
+#### Variables in okct-hello repo
+The following secrets are here named after environment (DEV) and the application it is connected to (OKCT_HELLO)
+* `secrets.DEV_REFERENCE_APP_ECR_PUSH_ROLE` - environment secret
+    * The full ARN for the role to assume when pushing the image, generated by terraform
+    * ex: `arn:aws:iam::123456789:role/github_actions/okctl-hello-github-ecr-push`
```
Comment (deifyed) [inline README.md]: okctl-hello references

File: README.md [hunk 2]
```diff
@@ -21,3 +21,38 @@ Run tests
...
+* `secrets.CLUSTER_DEPLOY_KEY` - repository secret
+    * See: https://docs.github.com/en/developers/overview/managing-deploy-keys
+    * See: https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#generating-a-new-ssh-key
+    * Note: keypair must be passwordless in order for the process to work
+    * Add public key for CLUSTER_DEPLOY_KEY to your IAC repo as a deploy key, remember to tick the "write access" box
+    * Add private key for CLUSTER_DEPLOY_KEY to your (this) hello-okcl repo as a repository secret
```
Comment (deifyed) [inline README.md]: okctl-hello reference

File: tf/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+# modules/iam/github_ecr_push
+Create resources for enabling a github account to push images to ECR
+
+Update [github_actions.tf](modules/iam/github_actions.tf) with your repo/account configuration
+
+Will create three resources in your account
+* `module.iam.aws_iam_openid_connect_provider.github_oidc`
+  * The OIDC provider that connects github with your account
+  * https://console.aws.amazon.com/iamv2/home#/identity_providers
+  * *NOTE*: the expiry date on the current `thumbprint_list` will expire, and needs to be updated when it does, there are no automatic update of the thumbprint list (see bin/generate_fingerprint_for_oidc.sh for script that is used to generate the thumbprint)
```
Comment (deifyed) [inline tf/README.md:27]: Do we have any plan / suggested way of handling this?

**Review by deifyed — APPROVED** [review]
LGTM! However I'd prefer it if @ivaruf also reviewed. I'm not that familiar with this code

File: tf/README.md (hunk 3 above)
Comment (haavardeide) [reply tf/README.md:27]: @deifyed no plan as of yet, that would include some jobs that sits and checks for expired/updated cert
The included script is a starting point for updating the terraform module easily

**Pull request merged**